### PR TITLE
Manage journald service and configuration

### PR DESCRIPTION
**Pull request by treydock**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -91,6 +98,8 @@
   Boolean                                                $manage_accounting,
   Hash[String,String]                                    $accounting,
   Boolean                                                $purge_dropin_dirs,
+  Boolean                                                $manage_journald,
+  Hash[String, String]                                   $journald_config,
```
Comment (bastelfreak) [inline manifests/init.pp]:
```suggestion
  Hash[String[1], String[1]]                                   $journald_config,
```

Comment (treydock) [reply manifests/init.pp]: Updated.

**Comment by trevor-vaughan** [comment]
No type validation? Are the `journald` options just too varied?

**Comment by bastelfreak** [comment]
Hey @treydock, thanks for the work! I did an inline suggestion. Can you also document it in the README.md?

**Comment by treydock** [comment]
@trevor-vaughan I suppose a custom type where the validation is done on a per-key basis would make sense, just not sure how it would survive for varied versions of systemd.  This would require subtypes to handle the varied units that are allowed for different keys.

@bastelfreak Updated README.

**Comment by trevor-vaughan** [comment]
@treydock Yeah, I get it. The config hashes just always feel like a 'slightly better than just flopping down raw content' approach to life without some sort of validation :-|

**Comment by raphink** [comment]
Agreed @trevor-vaughan. I usually like to provide a Data Type for the hash options to limit the scope of input data.

**Comment by treydock** [comment]
I'll add a custom data type.

**Comment by treydock** [comment]
Added custom data type but looks like #14 did same thing as this pull request.  Should I just make this pull request the addition of data type?

**Comment by raphink** [comment]
Ah right, I'm very sorry about that @treydock 	I hadn't noticed… Sure, the data type would still be a nice addition.

**Comment by ekohl** [comment]
https://rspec-puppet.com/documentation/type_aliases/ can be used to test type aliases for various input.

**Comment by raphink** [comment]
Very nice!

**Comment by treydock** [comment]
New data type added

**Pull request merged**